HW0/P4/P4.py

```diff
@@ -30,47 +30,123 @@
     # Load true trajectory and plot it
     # Normally, this data wouldn't be available in the real world
     #####################
-
-    # ax.plot(x_coords, y_coords, z_coords,
-    #         '--b', label='True trajectory')
-
+    s_truth = np.loadtxt('P4_trajectory.txt', delimiter=',', dtype=float)
+    x_coords = s_truth.T[0]
+    y_coords = s_truth.T[1]
+    z_coords = s_truth.T[2]
+
+    ax.plot(x_coords, y_coords, z_coords,
+             '--b', label='True trajectory')
+    
+    #plt.show()
     #####################
     # Part 2:
     #
     # Read the observation array and plot it (Part 2)
     #####################
-
-    # ax.plot(x_coords, y_coords, z_coords,
-    #         '.g', label='Observed trajectory')
-
+    measured_coord = np.loadtxt('P4_measurements.txt', delimiter=',', dtype=float)
+    
+    corr_matrix = np.zeros(shape=(3,3))
+    corr_matrix[0,0]=1/rx; corr_matrix[1,1]=1/ry; corr_matrix[2,2]=1/rz
+    measur_out = np.dot(corr_matrix, measured_coord.T)
+
+    xm_coords = measur_out[0]
+    ym_coords = measur_out[1]
+    zm_coords = measur_out[2]
+
+    ax.plot(xm_coords, ym_coords, zm_coords, '.g', label='Observed trajectory')
+    
     #####################
     # Part 3:
     # Use the initial conditions and propagation matrix for prediction
     #####################
 
     # A = ?
+    A = np.zeros(shape=(6,6)) + np.eye(6,6)
+    A[3][3]=1-c*dt; A[4][4]=1-c*dt; A[5][5]=1-c*dt
+    A[0][3]=dt; A[1][4]=dt; A[2][5]=dt
+
     # a = ?
+    a = np.zeros(6)
+    a[-1] = g * dt
+    
     # s = ?
+    s0 = np.array([0, 0, 2, 15, 3.5, 4]).T
 
     # Initial conditions for s0
+    s = np.zeros(shape=(6,K)) 
+    s[:,0] = s0
+    
     # Compute the rest of sk using Eq (1)
+    for j in xrange(K-1):
+        s[:,j+1] = np.dot(A, s[:,j]) + a
+
+    xb_coords = s[:,0:-1][0]
+    yb_coords = s[:,0:-1][1]
+    zb_coords = s[:,0:-1][2]
 
```

Comment:
You could just do xb_coords = s[0 , :] to get the same result. Basically just taking the whole first row. Not a big deal, just a bit cleaner.